### PR TITLE
js code generator enhancements

### PR DESCRIPTION
**Pull request by killanch**

### CODE REVIEW

**Comment by killanch** [comment]
@natabal pls review
generated code samples at: https://github.mv.usa.alcatel.com/Architect/vsd-react-ui/pull/80

File: monolithe/generators/lang/javascript/writers/apiversionwriter.py [hunk 1]
```diff
@@ -12,6 +16,8 @@ def __init__(self, monolithe_config, api_info):
         output = monolithe_config.get_option("output", "transformer")
 
         self.output_directory = "%s/javascript/%s" % (output, api_info["version"])
+        self.enum_directory =  "%s/enums" % self.output_directory
```
Comment (natabal) [inline monolithe/generators/lang/javascript/writers/apiversionwriter.py]: what about the abstract folder? Do you need to make sure that gets create as well?

**Comment by natabal** [comment]
@ikarthik I don't see where you are creating the abstract classes. This should be generated as well if it does not exist already.

**Comment by killanch** [comment]
@natabal added code to generate AbstractNamedEntity
Another related query, today the directory structure of generated content is,
codegen/javascript/5.0/
codegen/javascript/5.0/abstract
codegen/javascript/5.0/enums

Is this okay? Or do we want it to be something like,
codegen/javascript/5.0/models
codegen/javascript/5.0/models/abstract
codegen/javascript/5.0/models/enums

Pls let me know.

**Comment by natabal** [comment]
@killanch let's have this:

codegen/javascript/5.0/models
codegen/javascript/5.0/models/abstract
codegen/javascript/5.0/models/enums

As well in models please create an index.js and export all the models along with the AbstractNamedEntity, similar with enums/index.js.

This way we just need to import them as * from 'models'
Thanks!

**Comment by killanch** [comment]
@natabal changes done. added index for model. im pushing in the changes.

**Pull request merged**